benchmarks/src/replace.cpp

```diff
@@ -0,0 +1,73 @@
+    "volutpat nunc sit amet accumsan tempor. Quisque pretium vestibulum ultricies. Suspendisse potenti. Aenean at diam "
+    "iaculis, condimentum felis venenatis, condimentum erat. Nam quis elit dui. Duis quis odio vitae metus hendrerit "
+    "rhoncus ut et magna. Cras ac augue quis nibh pharetra sagittis. Donec ullamcorper vel eros semper pretium. Proin "
+    "vel sollicitudin eros. Nulla sollicitudin mattis turpis id suscipit. Aliquam sed risus velit. Aliquam iaculis nec "
+    "nibh ac egestas. Duis finibus semper est sed consequat. Sed in sapien quis nibh dignissim mattis. Vestibulum nec "
+    "metus sodales, euismod mauris ac, sollicitudin libero. Maecenas non arcu ac velit ullamcorper fringilla et quis "
+    "nulla. Curabitur posuere leo eget ipsum tincidunt dignissim. Cras ultricies suscipit neque, quis suscipit tortor "
+    "venenatis non. Cras nisl mi, bibendum in vulputate quis, vestibulum ornare enim. Nunc hendrerit placerat dui, "
+    "aliquam mollis sem convallis et. Integer vitae urna diam. Phasellus et imperdiet est. Maecenas auctor facilisis "
+    "nibh non commodo. Suspendisse iaculis quam id bibendum feugiat. Pellentesque felis erat, egestas a libero ac, "
+    "laoreet consectetur elit. Cras ut suscipit ex. Etiam gravida sem quis ex porta, eu lacinia tortor fermentum. "
+    "Nulla consequat odio enim, sed condimentum est sagittis a. Quisque nec commodo tellus. Phasellus elementum "
+    "feugiat dolor et feugiat. Praesent sed mattis tortor. In vitae sodales purus. Morbi accumsan, ligula et interdum "
+    "lacinia, leo risus suscipit urna, non luctus mi justo eu ipsum. Curabitur venenatis pretium orci id porttitor. "
+    "Quisque dapibus nisl sit amet elit lobortis sagittis. Orci varius natoque penatibus et magnis dis parturient "
+    "montes, nascetur ridiculus mus. Mauris varius dui sit amet tortor facilisis vestibulum. Curabitur condimentum "
+    "justo nec orci mattis auctor. Quisque aliquet condimentum arcu ac sollicitudin. Maecenas elit elit, condimentum "
+    "vitae auctor a, cursus et sem. Cras vehicula ante in consequat fermentum. Praesent at massa nisi. Mauris pretium "
+    "euismod eros, ut posuere ligula ullamcorper id. Nullam aliquam malesuada est at dignissim. Pellentesque finibus "
+    "sagittis libero nec bibendum. Phasellus dolor ipsum, finibus quis turpis quis, mollis interdum felis.";
+
+template <class T>
+void rc(benchmark::State& state) {
+    std::vector<T> a(std::size(src));
+    std::vector<T> b(std::size(src));
+
+    std::copy(std::begin(src), std::end(src), std::begin(a));
+
+    for (auto _ : state) {
+        std::replace_copy(std::begin(a), std::end(a), std::begin(b), T{'m'}, T{'w'});
+    }
+}
+
+template <class T>
+void rc_if(benchmark::State& state) {
+    std::vector<T> a(std::size(src));
+    std::vector<T> b(std::size(src));
+
+    std::copy(std::begin(src), std::end(src), std::begin(a));
+
+    for (auto _ : state) {
+        std::replace_copy_if(

```

Comment:
While it's not strictly necessary, I'd recommend void-casting `std::replace_copy_if()` here, as a reminder that we don't care if we're partially overwriting `b`. (We haven't marked the algorithm as `[[nodiscard]]` because calling it with a `back_insert_iterator` etc. makes the return value irrelevant, but in general if you're writing to a plain iterator and you don't know how many values are going to satisfy the predicate, you really should be interested in the return value.)